src/main/webapp/user-page.html

```diff
@@ -16,29 +16,67 @@
 
 <!DOCTYPE html>
 <html>
+  <head>
+    <title>User Page</title>
+    <meta charset="UTF-8">
+    <link rel="stylesheet" href="/css/main.css">
+    <link rel="stylesheet" href="/css/user-page.css">
+    <script src="/js/user-page-loader.js"></script>
+  </head>
 
-<head>
-  <title>User Page</title>
-  <meta charset="UTF-8">
-  <link rel="stylesheet" href="/css/main.css">
-  <link rel="stylesheet" href="/css/user-page.css">
-  <script src="/js/user-page-loader.js"></script>
-</head>
-
-<body onload="buildUI();">
-  <nav>
-    <ul id="navigation">
-      <li><a href="/">Home</a></li>
-    </ul>
-  </nav>
-  <h1 id="page-title">User Page</h1>
-
-  <form id="message-form" action="/messages" method="POST" class="hidden">
-    Enter a new message:
-    <br/>
-    <textarea name="text" id="message-input"></textarea>
-    <br/>
-    <input type="submit" value="Submit">
+  <body onload="buildUI();">
+    <nav>
+      <ul id="navigation">
+        <li><a href="/">Home</a></li>
+      </ul>
+    </nav>
+    <h1 id="page-title">User Page</h1>
+   
+    <jsp:directive.taglib uri="http://java.sun.com/jsp/jstl/core" prefix="c"%>
+    <jsp.directive.taglib uri="http://java.sun.com/jsp/jstl/functions" prefix="fn"%>
+<div class="userForm">
+  <h2>
+    Input what tea you drank today
+  </h2>
+
+  <form method="POST" action="/api/user-form-data">
```

Comment:
I would remove the `method` and `action` properties and, instead, handle form submission in JavaScript. To handle in JavaScript, change:

```
<button type="submit">Save</button>
```

To:

```
<button onclick='onSubmit'>Save</button>
```

And then define a function called `onSubmit` in your JavaScript code.

An alternative approach is to create a CSS selector on the button and then register the action in JavaScript:

```
// JavaScript
var submitButton = document.querySelector('#submit-button');
submitButton.addEventListener('click', onSubmit);

<!-- HTML -->
<button id="submit-button">Save</button>
```

Reply:
So, one user on stackoverflow mention that one of the advantages of using a submit button was if a user doesn't have JavaScript enabled in their browser, the form would still work. This is why right now I have it where JavaScript is not handling the form submission yet. 

Should I just assume that in our scope, that all potential users will have JavaScript enabled within their browsers and that they wouldn't run into this issue?

https://stackoverflow.com/questions/16050798/html-form-submit-using-javascript-vs-submit-button

Reply:
Yes, this is accurate and known as "graceful degradation"; web apps that need to target browsers with and without JavaScript will often have a handler in JavaScript (which can provide more immediate validation and feedback without issuing a network request) as well as an action that points to a server-side submit form that is invoked in the JavaScript-less case.

That being said, JavaScript is supported in all modern browsers, and most modern websites assume that JavaScript is supported (and often will not work without JavaScript). Given that most of the other code already implemented has a dependency on JavaScript with no graceful degradation (e.g. the login functionality is implemented in a such a way that JavaScript is required), I would not spend a whole lot of time implementing graceful degradation.

In a real application, it is typical to first target the vast majority of cases (assume a modern browser, assume JavaScript is supported, etc., etc.) and only then, later, after getting this minimum viable product working, look to extend support to older browsers or browsers that have JavaScript disabled.

Reply:
Got it, it would be a headache to implement a fallback for the users if we already created code that already assumes that the user will have JavaScript enabled. 

Thank you for the comment! It's interesting to read about the difference between "graceful degradation" and "progressive enhancement".